storm-server/src/main/java/org/apache/storm/localizer/LocalizedResourceRetentionSet.java

```diff
@@ -94,17 +93,17 @@ public void cleanup(ClientBlobStore store) {
             Map.Entry<LocallyCachedBlob, Map<String, ? extends LocallyCachedBlob>> rsrc = i.next();
             LocallyCachedBlob resource = rsrc.getKey();
             try {
-                resource.getRemoteVersion(store);
+                if (!store.isRemoteBlobExists(resource.getKey())) {
```

Comment:
Admittedly the code is cleaner with this, but the change is totally unneeded.  It behaves exactly the same as it did before.  I think this is a good change, it would just be nice to have it be a separate pull request and a separate JIRA as it is not really a part of the needed fix.

Reply:
Ok, i will separate it apart with another JIRA.